src/blockchain/mod.rs

```diff
@@ -254,6 +253,12 @@ impl<K: KvStore> KvStoreChain<K> {
     ) -> Result<Vec<Transaction>, BlockchainError> {
         let mut sorted = txs.clone();
         sorted.sort_by(|t1, t2| t1.nonce.cmp(&t2.nonce));
+        sorted.sort_by(|t1, t2| {
```

Comment:
Hmm, any reasons for that?

Reply:
I see your point. In my new updates, treasury transactions can only happen in the genesis block, or as the first tx of any non-genesis block (as miner reward).

Reply:
the mempool txs, should NOT have any transaction with src="Treasury". So i would say, `drain()` them instead of sorting them :) or return an error like "TreasuryNotAllowedInMempool"